### ISSUE TITLE
Module was compiled with an incompatible version of Kotlin

### ISSUE THREAD

**Issue by rwinch** [reactor/reactor-core#2793]
Compiling Spring Security 5.3 with the latest version of reactor core produces the following error

```
git clone https://github.com/spring-projects/spring-security
cd spring-security
git checkout -b 5.3.x origin/5.3.x
./gradlew test -PforceMavenRepositories=snapshot -PspringVersion='5.2.+' -PreactorVersion=Dysprosium-BUILD-SNAPSHOT -PspringDataVersion=Lovelace-BUILD-SNAPSHOT
...
e: /home/rwinch/.gradle/caches/modules-2/files-2.1/io.projectreactor/reactor-core/3.3.21.BUILD-SNAPSHOT/5cbe0408c04f864de54b1c0d38e6f185c7caed0e/reactor-core-3.3.21.BUILD-SNAPSHOT.jar!/META-INF/reactor-core.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
```

It works just fine using the last release of reactor-core

```
./gradlew test -PforceMavenRepositories=snapshot -PspringVersion='5.2.+' -PreactorVersion=Dysprosium-SR23 -PspringDataVersion=Lovelace-BUILD-SNAPSHOT
```

## Expected Behavior

It should continue to work with the existing version of Kotlin

## Actual Behavior

Fails with

```
e: /home/rwinch/.gradle/caches/modules-2/files-2.1/io.projectreactor/reactor-core/3.3.21.BUILD-SNAPSHOT/5cbe0408c04f864de54b1c0d38e6f185c7caed0e/reactor-core-3.3.21.BUILD-SNAPSHOT.jar!/META-INF/reactor-core.kotlin_module: 
Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.16.
```

## Steps to Reproduce

```
git clone https://github.com/spring-projects/spring-security
cd spring-security
git checkout -b 5.3.x origin/5.3.x
./gradlew test -PforceMavenRepositories=snapshot -PspringVersion='5.2.+' -PreactorVersion=Dysprosium-BUILD-SNAPSHOT -PspringDataVersion=Lovelace-BUILD-SNAPSHOT
```

## Possible Solution

Either revert the version of Kotlin or specify the language and api versions.

## Your Environment

```
-----------------------------------------------------------
Gradle 6.8.3
------------------------------------------------------------

Build time:   2021-02-22 16:13:28 UTC
Revision:     9e26b4a9ebb910eaa1b8da8ff8575e514bc61c78

Kotlin:       1.4.20
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          1.8.0_275 (AdoptOpenJDK 25.275-b01)
OS:           Linux 5.4.0-86-generic amd64
```

* Reactor version(s) used: Dysprosium-BUILD-SNAPSHOT

**Comment by simonbasle** [comment]
Looks like I shouldn't have upgraded the kotlin gradle plugin version to 1.5.x, woops.
That was done along Gradle upgrade to 7.2.

One remediation that seems to work is to keep the newest plugin (to ensure compatibility of the build toolchain) but indicate that the kotlin `languageVersion` and `apiVersion` should be kept at `1.3` (even though these are deprecated):

```groovy
kotlinOptions {
	jvmTarget = "1.8"
	freeCompilerArgs = ["-Xjsr305=strict"]
	languageVersion = "1.3" //TODO kotlin languageVersion 1.3  is now deprecated
	apiVersion = "1.3"
}
```

I was able to verify that the classes generated were at metadata level `1.1.18` by using IntelliJ's "Tools > Kotlin > Show Kotlin Bytecode" menu/view. Clicking "decompile" on eg. `FluxExtensionKt.class` shows:

```java
@Metadata(
   mv = {1, 1, 18},
   bv = {1, 0, 3},
...
```

This is not `1.1.16` like in the message, but after some trouble getting `spring-security` to pick up a snapshot from my local maven repository, I was able to validate the command you provided @rwinch does start running tests. (where it would fail at compile time previously)
